scss/mixins/_pagination.scss

```diff
@@ -11,17 +11,66 @@
 
 // scss-docs-start pagination-max-items-mixin
 // Boosted mod
-// Ensure we only display `n` items:
-// - the first `n/2` items
-// - the last `n/2` items
-// - including the .active one (so maybe not the `n/2`th last item)
+// The pagination list ((m) items) should display exactly (n) items and is built in 3 parts:
+// - the begining of the list is made of the first (n/2) items
+// - the end of the list is made of the last (n/2) items (or (n/2 - 1) depending on the active element)
+// - the last part of the list is made of the active element
 @mixin pagination-max-items($pagination-max-items) {
-  display: none;
+  @if $enable-responsive-pagination {
+    $half-items: $pagination-max-items * .5;
+    display: none;
 
-  &.active,
-  &:nth-child(-n+#{$pagination-max-items * .5}),
-  &:nth-last-child(-n+#{$pagination-max-items * .5}) {
-    display: list-item;
+    // rule 0:
+    //  - adding the (n/2) first items
+    //  - adding the (n/2 - 1) last items
+    //  - adding the (n/2) last item if the list contains exactly (n) items
+    &:nth-child(-n+#{$half-items}),
+    &:nth-last-child(-n+#{$half-items - 1}),
+    &:nth-child(#{$half-items + 1}):nth-last-child(#{$half-items}) {
+      display: flex;
+    }
+
+    // Pagination with m > n:
+    //  - rule 1: adding 3 dots after the active element if situated between the first (n/2) items and the last (n/2 - 1) items
+    //            no dots added if the active element is exactly the (n/2 + 1)th first item and exactly the (n/2)th last item
+    //  - rule 2: adding 3 dots after the (n/2)th first element of the list
+    &:nth-child(#{$half-items}) ~ .active:not(:nth-last-child(-n+#{$half-items - 1})):not(:nth-child(#{$half-items + 1}):nth-last-child(-n+#{$half-items})),
+    &:nth-child(#{$half-items}):not(:nth-last-child(-n+#{$half-items + 1})) {
+      &::after {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        width: calc((var(--#{$prefix}pagination-icon-size) + var(--#{$prefix}pagination-border-width) * 2)); // stylelint-disable-line function-disallowed-list
+        content: "...";
+        background-color: var(--#{$prefix}pagination-bg);
+      }
+    }
+
+    // Pagination with m > n:
+    //  - rule 3: hiding 3 dots of the rule 2 above if the active element is exactly the (n/2 + 1)th first item
```

Comment:
Would it be good and/or necessary to explain theses rules in the documentation too? Especially if users would like to customize the pagination in their ptoject?